### PR TITLE
Onboarding the rosdep to ROS2 Rolling distro

### PR DESCRIPTION
**Pull request by seanyen**
This pull request is to onboard the rosdep to ROS2 Rolling distro.

* Adding a patch.bat to capture the process of injecting the rosdep to the installation.
* Adding `msft_ros2_env` package to inject the missing environment variable `ROS_ETC_DIR` in ROS2.
* Copying `windows.yaml` from Noetic distro for bootstrap.

### CODE REVIEW

File: ros/rolling/patch/etc/ros/rosdep/sources.list.d/10-ms-iot.list [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# os-specific listings first
+yaml file://localhost/c:\opt\ros\rolling\x64\etc\ros\rosdep\windows.yaml windows
```
Comment (ooeygui) [inline ros/rolling/patch/etc/ros/rosdep/sources.list.d/10-ms-iot.list:2]: Is this correct? It looks really odd. Shouldn't it be something like `file:///c:/`

File: ros/rolling/src/msft_ros2_env/package.xml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<package format="2">
+  <name>msft_ros2_env</name>
+  <version>0.0.1</version>
+  <description>The package provides the environment variables useful for ROS on Windows.</description>
+  <maintainer email="seanyen@microsoft.com">Sean Yen</maintainer>
```
Comment (ooeygui) [inline ros/rolling/src/msft_ros2_env/package.xml:5]: Can you make this riotsquad@microsoft.com?

**Review by ooeygui — APPROVED** [review]

File: ros/rolling/src/msft_ros2_env/package.xml (hunk 2 above)
Comment (seanyen) [reply ros/rolling/src/msft_ros2_env/package.xml]: updated.

File: ros/rolling/patch/etc/ros/rosdep/sources.list.d/10-ms-iot.list (hunk 1 above)
Comment (seanyen) [reply ros/rolling/patch/etc/ros/rosdep/sources.list.d/10-ms-iot.list]: Both work in Python. But I changed it to `file:///` which seems to be more commonly seen.

**Pull request merged**